library/lib/application.cpp

```diff
@@ -456,6 +588,15 @@ void Application::navigate(FocusDirection direction)
     if (!currentFocus || !currentFocus->hasParent())
         return;
 
+    // The item was "loosely" unfocused, i.e. we just wanted the highlight
+    // to disappear. Now we need it to reappear because the gamepad is
+    // navigating
+    // Almost like HOS, its not perfect.
```

Comment:
How is it "almost" like HOS? What's the difference?

Reply:
Just the way it currently behaves.
When navigating by gamepad in HOS and then switching over to navigating by touch-screen causes the "focus" to disappear. Then when switch back to gamepad navigation by pressing "OK" (The 'A' button) causes the focus to reappear, well pressing any button makes that happen.
However in this draft, when switching back to gamepad navigation by pressing "OK" causes the onclick event to fire. Thus, the behavior is "almost" like HOS.

Reply:
Okay I see. I think the way it's implemented in the draft is actually bad design - people won't expect it to behave that way and it will cause troubles. It's the kind of little things that make you hate a piece of software as a user 😬

Reply:
I think the best way to approach this is to implement some type of "Navigation Controller," where navigation works a certain way based on the input method (gamepad/keyboard or touchscreen.)

Reply:
Can't you simply implement the missing behavior? With a touch mode boolean for instance. When pressing a button if in touch mode, don't call the click event but simply disable touch mode and show the focus back ?